From 74aab3fda289a65cab84b0dfaec3662bb4fb2e52 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:26 +0800
Subject: [PATCH] arm64/mpam: fix hard code address map for 1620 2P
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 74aab3fda289a65cab84b0dfaec3662bb4fb2e52


hulk inclusion
category: bugfix
bugzilla: 48265
CVE: NA

--------------------------------

The base address is hard code in current version.
P1's base address changed from 0x4000xxxxxxxx to 0x200000xxxxxx.
We need adjust it, or failed to enable mpam.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index eb9e38284bc7..bd94cd31c2fa 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -74,8 +74,8 @@ char *mpam_types_str[] = {
  * ------------------------------------
  *   01 |    0 | P0 TB | 0x000098xxxxxx
  *   03 |    1 | P0 TA | 0x000090xxxxxx
- *   05 |    2 | P1 TB | 0x400098xxxxxx
- *   07 |    3 | P2 TA | 0x400090xxxxxx
+ *   05 |    2 | P1 TB | 0x200098xxxxxx
+ *   07 |    3 | P2 TA | 0x200090xxxxxx
  *
  *   AFF2: MPIDR.AFF2
  */
@@ -92,13 +92,13 @@ char *mpam_types_str[] = {
 struct mpam_node mpam_node_all[] = {
 	MPAM_NODE(L3TALL0, MPAM_RESOURCE_CACHE, 0x000098ULL, 0xB9),
 	MPAM_NODE(L3TALL1, MPAM_RESOURCE_CACHE, 0x000090ULL, 0xB9),
-	MPAM_NODE(L3TALL2, MPAM_RESOURCE_CACHE, 0x400098ULL, 0xB9),
-	MPAM_NODE(L3TALL3, MPAM_RESOURCE_CACHE, 0x400090ULL, 0xB9),
+	MPAM_NODE(L3TALL2, MPAM_RESOURCE_CACHE, 0x200098ULL, 0xB9),
+	MPAM_NODE(L3TALL3, MPAM_RESOURCE_CACHE, 0x200090ULL, 0xB9),
 
 	MPAM_NODE(HHAALL0, MPAM_RESOURCE_MC, 0x000098ULL, 0xC1),
 	MPAM_NODE(HHAALL1, MPAM_RESOURCE_MC, 0x000090ULL, 0xC1),
-	MPAM_NODE(HHAALL2, MPAM_RESOURCE_MC, 0x400098ULL, 0xC1),
-	MPAM_NODE(HHAALL3, MPAM_RESOURCE_MC, 0x400090ULL, 0xC1),
+	MPAM_NODE(HHAALL2, MPAM_RESOURCE_MC, 0x200098ULL, 0xC1),
+	MPAM_NODE(HHAALL3, MPAM_RESOURCE_MC, 0x200090ULL, 0xC1),
 };
 
 int mpam_nodes_init(void)
-- 
2.26.2

